pkg/geo/geomfn/swap_ordinates_test.go

```diff
@@ -0,0 +1,220 @@
+			input2: "yx",
+			want:   geom.NewPointFlat(geom.XY, []float64{20.2, 20.2}),
+		},
+		{
+			name:   "swap ordinate 2D point XY(capitalize)  with the same X and Y value",
+			input1: geom.NewPointFlat(geom.XY, []float64{20.2, 20.2}),
+			input2: "Yx",
+			want:   geom.NewPointFlat(geom.XY, []float64{20.2, 20.2}),
+		},
+		{
+			name:   "swap x to y from ordinate 2D POINT",
+			input1: geom.NewPointFlat(geom.XY, []float64{10.2, 20.2}),
+			input2: "xy",
+			want:   geom.NewPointFlat(geom.XY, []float64{20.2, 10.2}),
+		},
+		{
+			name:   "swap y to x from ordinate 2D Line String",
+			input1: geom.NewLineStringFlat(geom.XY, []float64{1, 2, 3, 4}),
+			input2: "yx",
+			want:   geom.NewLineStringFlat(geom.XY, []float64{2, 1, 4, 3}),
+		},
+		{
+			name:   "swap y to x from ordinate 2D Line String with same coordinates",
+			input1: geom.NewLineStringFlat(geom.XY, []float64{2, 2, 3, 3}),
+			input2: "yx",
+			want:   geom.NewLineStringFlat(geom.XY, []float64{2, 2, 3, 3}),
+		},
+		{
+			name:   "swap y to x from ordinate 2D Polygon",
+			input1: geom.NewPolygonFlat(geom.XY, []float64{0, 0, 5.55, 5.55, 0, 10, 0, 0}, []int{8}),
+			input2: "yx",
+			want:   geom.NewPolygonFlat(geom.XY, []float64{0, 0, 5.55, 5.55, 10, 0, 0, 0}, []int{8}),
+		},
+		{
+			name:   "swap y to x from coordinates of a multi polygon",
+			input1: geom.NewMultiPolygonFlat(geom.XY, []float64{0, 0, 4, 0, 4, 4, 0, 4, 0, 0, 1, 1, 2, 1, 2, 2, 1, 2, 1, 1, -1, -1, -1, -2, -2, -2, -2, -1, -1, -1}, [][]int{{10, 20}, {30}}),
+			input2: "yx",
+			want:   geom.NewMultiPolygonFlat(geom.XY, []float64{0, 0, 0, 4, 4, 4, 4, 0, 0, 0, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, -1, -1, -2, -1, -2, -2, -1, -2, -1, -1}, [][]int{{10, 20}, {30}}),
+		},
+		{
+			name:   "swap y to x from coordinates of a multi line string",
+			input1: geom.NewMultiLineStringFlat(geom.XY, []float64{0, 0, 1, 1, 1, 2, 2, 3, 3, 2, 5, 4}, []int{6, 12}),
+			input2: "yx",
+			want:   geom.NewMultiLineStringFlat(geom.XY, []float64{0, 0, 1, 1, 2, 1, 3, 2, 2, 3, 4, 5}, []int{6, 12}),
+		},
+		{
+			name:   "swap y to x from coordinates of an empty line string",
+			input1: geom.NewLineString(geom.XY),
+			input2: "yx",
+			want:   geom.NewLineString(geom.XY),
+		},
+		{
+			name:   "swap y to x from coordinates of an empty polygon",
+			input1: geom.NewPolygon(geom.XY),
+			input2: "yx",
+			want:   geom.NewPolygon(geom.XY),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			geometry, err := geo.MakeGeometryFromGeomT(tt.input1)
+			require.NoError(t, err)
+
+			got, err := SwapOrdinates(geometry, tt.input2)
+			if (err != nil) && (tt.errString != err.Error()) {
+				t.Errorf("SwapOrdinates() error = %v, wantErr %v", err, tt.errString)
+				return
+			}
+
+			geometryWant, err := geo.MakeGeometryFromGeomT(tt.want)
+			require.NoError(t, err)
+			require.Equal(t, geometryWant, got)
+		})
+	}
+}
+
+func TestSwapOrdinates_Collection(t *testing.T) {
+	tests := []struct {
```

Comment:
we normally don't separate test cases like this. this seems easy to fold with the above test cases (i.e. reuse the same struct)? *geom.Collection can be used as geom.T, so you can just extend the number of inputs.